server/src/option.rs

```diff
@@ -606,4 +605,22 @@ pub mod validation {
     pub fn url(s: &str) -> Result<url::Url, String> {
         url::Url::parse(s).map_err(|_| "Invalid URL provided".to_string())
     }
+
+    pub fn size(s: &str) -> Result<(), String> {
+        if let Ok(re) = regex::Regex::new(SIZE_WITH_UNIT_REGEX) {
+            if re.captures(s).is_some() {
+                Ok(())
+            } else {
+                Err(
+                    "Invalid size provided. Please provide size in the format '1GiB', '3MiB', etc"
+                        .to_string(),
+                )
+            }
+        } else {
+            Err(
+                "Invalid size provided. Please provide size in the format '1GiB', '3MiB', etc"
+                    .to_string(),
+            )
+        }
```

Comment:
```suggestion
    pub fn size(s: &str) -> Result<(), String> {
        // Unwrapping is Ok because the static expression is valid.
        let regex = regex::Regex::new(r"^(\d+)([TGMKP][iI]?[Bb]?)$").unwrap();

        if regex.is_match(s) {
            return Ok(());
        }

        return Err(
            "Invalid size provided. Please provide size in the format '1GiB', '3MiB', etc"
                .to_owned(),
        );
```